api/v1alpha3/cluster_types.go

```diff
@@ -57,6 +62,18 @@ type ClusterSpec struct {
 
 // ANCHOR_END: ClusterSpec
 
+// ANCHOR: PostApplyAddon
+
+// PostApplyAddon specifies the addon's Secret parameters.
+type PostApplyAddon struct {
+	// Name is the name of the secret.
+	Name string `json:"name,omitempty"`
+	// Namespace is the namespace of the secret.
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
These can probably be in a field called `SecretRef` of type `corev1.ObjectReference` instead of separate fields.

Reply:
I'd like to avoid any new use of `ObjectReference` if we can avoid it at all. I think we should favor `corev1.LocalObjectReference` or `corev1.TypedLocalObjectReference` as long as we can avoid the need for a namespace. If we need a namespace, then we should probably think about creating a new type that is a subset of `ObjectReference`